src/EnergyPlus/InputProcessing/IdfParser.cc

```diff
@@ -497,24 +499,19 @@ json IdfParser::parse_number(std::string const &idf, size_t &index, bool &succes
 
 json IdfParser::parse_value(std::string const &idf, size_t &index, bool &success, json const &field_loc)
 {
+    Token token;
     auto const &field_type = field_loc.find("type");
     if (field_type != field_loc.end()) {
         if (field_type.value() == "number" || field_type.value() == "integer") {
-            return parse_number(idf, index, success);
+            token = Token::NUMBER;
```

Comment:
It is more than just that. Originally there was duplicate code for parsing number or string depending if it found "type" or not. So it first looked for "type" and would then parse directly based on that type in the schema. If it didn't find a "type" then it would try to determine if it was a number or string and parse it. The code for parsing number or string was different in these two approaches subtly. This refactors the code to determine if it is probably a string or number first then there is the code to parse number or string. So now both approaches use the more robust number and string parsing code.

Reply:
:+1: thanks for the clarification!